### PR TITLE
Move the disabling of samples and operatorhub sources earlier in the installation process

### PR DESCRIPTION
**Pull request by bennerv**
### Which issue this PR addresses:

Fixes [ADO#15431594](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/15431594)

### What this PR does / why we need it:

There's a issue where the clusterversion CR waits for samples to go healthy.  If we disable samples before we wait for the clusterversion, then it should not error out in installations.  

ICM 331555133 has a similar failure scenario

### Test plan for issue:

I can't replicate this, but customers are experiencing this.  The clusterversion CR shouldn't complain about a disabled operator so it's moved earlier in the install process.  

### Is there any documentation that needs to be updated for this PR?

no

### CODE REVIEW

**Review by dem4gus — APPROVED** [review]
lgtm

**Comment by bennerv** [comment]
> lgtm. Is it possible to also disable the BareMetal operator the same way or do we just rely on it disabling itself?

It's possible in 4.11 now I guess
https://docs.openshift.com/container-platform/4.11/release_notes/ocp-4-11-release-notes.html#ocp-4-11-capabilities

So soon :)

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -247,11 +247,11 @@ func (m *manager) Install(ctx context.Context) error {
 			steps.Condition(m.operatorConsoleExists, 30*time.Minute, true),
 			steps.Action(m.updateConsoleBranding),
 			steps.Condition(m.operatorConsoleReady, 20*time.Minute, true),
+			steps.Action(m.disableSamples),
+			steps.Action(m.disableOperatorHubSources),
 			steps.Condition(m.clusterVersionReady, 30*time.Minute, true),
 			steps.Condition(m.aroDeploymentReady, 20*time.Minute, true),
 			steps.Action(m.disableUpdates),
```
Comment (cblecker) [inline pkg/cluster/install.go:254]: We should also probably move updates up?

Comment (bennerv) [reply pkg/cluster/install.go:254]: disableUpdate just removes the channel from the clusterversion CR.  I was mostly focused on things that could impact the clusterversion status on installation.

Comment (cblecker) [reply pkg/cluster/install.go:254]: It will speed it up if it doesn't need to pull cincinnati updates

Comment (bennerv) [reply pkg/cluster/install.go]: updated - ty for the info.  TIL

**Review by cblecker — APPROVED** [review]
lgtm

**Pull request merged**